cumulus/parachains/integration-tests/emulated/tests/assets/asset-hub-westend/src/tests/reserve_transfer.rs

```diff
@@ -1132,8 +1145,666 @@ fn reserve_transfer_native_asset_from_para_to_para_through_relay() {
+	// TODO: Also allow payment with different assets locally, right now only works remotely.
+	// <AssetHubWestend as TestExt>::execute_with(|| {
+	// 	assert_ok!(<AssetHubWestend as AssetHubWestendPallet>::Balances::force_set_balance(
+	// 		<AssetHubWestend as Chain>::RuntimeOrigin::root(),
+	// 		sender.clone().into(),
+	// 		ASSET_HUB_WESTEND_ED,
+	// 	));
+	// });
+
+	let para_test_args = TestContext {
+		sender: sender.clone(),
+		receiver: receiver.clone(),
+		args: TestArgs::new_para(
+			destination,
+			receiver.clone(),
+			asset_amount_to_send,
+			assets,
+			None,
+			fee_asset_index,
+		),
+	};
+	let mut test = SystemParaToParaTest::new(para_test_args);
+
+	// Query initial balances.
+	let sender_balance_before = test.sender.balance;
+	let sender_assets_before = AssetHubWestend::execute_with(|| {
+		type Assets = <AssetHubWestend as AssetHubWestendPallet>::Assets;
+		<Assets as Inspect<_>>::balance(asset_id, &sender)
+	});
+	let receiver_foreign_assets_before = PenpalA::execute_with(|| {
+		type ForeignAssets = <PenpalA as PenpalAPallet>::ForeignAssets;
+		<ForeignAssets as Inspect<_>>::balance(
+			system_para_foreign_asset_location.clone(),
+			&receiver,
+		)
+	});
+
+	// Set assertions and dispatchables
+	// TODO: Bring back assertions.
+	// test.set_assertion::<AssetHubWestend>(system_para_to_para_assets_sender_assertions);
+	// test.set_assertion::<PenpalA>(system_para_to_para_assets_receiver_assertions);
+	test.set_dispatchable::<AssetHubWestend>(system_para_to_para_reserve_transfer_assets);
+	test.assert();
+
+	// Query initial balances.
+	let sender_balance_after = test.sender.balance;
+	let sender_assets_after = AssetHubWestend::execute_with(|| {
+		type Assets = <AssetHubWestend as AssetHubWestendPallet>::Assets;
+		<Assets as Inspect<_>>::balance(asset_id, &sender)
+	});
+	let receiver_foreign_assets_after = PenpalA::execute_with(|| {
+		type ForeignAssets = <PenpalA as PenpalAPallet>::ForeignAssets;
+		<ForeignAssets as Inspect<_>>::balance(
+			system_para_foreign_asset_location.clone(),
+			&receiver,
+		)
+	});
+
+	// TODO: When we allow payment with different assets locally, this should be the same, since
+	// they aren't used for fees.
+	assert!(sender_balance_before > sender_balance_after);
+	// The assets have been sent and used for fees.
+	assert!(sender_assets_after < sender_assets_before - asset_amount_to_send);
+	assert!(receiver_foreign_assets_after > receiver_foreign_assets_before);
+}
+
+// ================================================================================
+// ===== Reserve Transfers - Sufficient Asset - Parachain->AssetHub->Parachain ====
+// ================================================================================
+/// Reserve Transfers of a sufficient trust-backed asset from Parachain to Parachain (through Relay
+/// reserve) should work
+#[test]
+fn reserve_transfer_sufficient_asset_from_para_to_para_through_asset_hub() {
```

Comment:
It's a bit confusing to use USDT in multiple tests, sometimes buy through pool, sometimes through direct weight-trader.

I would remove this test and keep only `reserve_transfer_pool_assets_from_para_to_para_through_asset_hub` - that is the "real world use case" one.

Also, "sufficient" (now) only means that asset can cover ED - not that it can be used to buy weight directly - we should be careful to not propagate this misconception that sufficient means it can pay for weight directly 🙈

Reply:
Sufficient means that it only covers ED but right now it also means we swap it. That's the idea that I had. In case we no longer need to support that, I can just drop all the support for paying fees with sufficient assets from this PR.

Reply:
yes, I would drop it completely

at most keep it on some archive branch if we ever find a usecase for it in the future, but based on our current knowledge pools everywhere is the plan :))

Reply:
That will greatly simplify this PR then, will remove